lib/protractor.js

```diff
@@ -698,14 +699,7 @@ Protractor.prototype.initDebugger_ = function(debuggerClientPath, opt_debugPort)
       self.execPromiseResult_ = self.execPromiseError_ = undefined;
 
       self.execPromise_ = self.execPromise_.
-          then(function() {
-            var result = execFn_();
-            if (webdriver.promise.isPromise(result)) {
-              return result.then(function(val) {return val;});
```

Comment:
`element(by.xyz).getText()` is an elementFinder/promise.
When someone enters `element(by.xyz).getText()`, we need to return the text value of the elementFinder (i.e. by calling then) instead of the text representation of the elementFinder.